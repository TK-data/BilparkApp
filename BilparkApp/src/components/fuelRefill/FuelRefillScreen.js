import React from 'react';
import { StyleSheet, View } from 'react-native';
import { Content } from 'native-base';
import { connect } from 'react-redux';

import SettingsButton from './SettingsButton';
import { getFuelRefills } from '../../actions/fuelRefill';
import FuelRefillForm from './FuelRefillForm';
import FuelRefillList from './FuelRefillList';
import FuelSetNotificationScreen from '../fuelDayModal/FuelSetNotificationScreen';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#002776',
  },
  modal: {
    alignContent: 'center',
  },
  content: {
    flex: 2,
    justifyContent: 'flex-start',
    backgroundColor: '#002776',
  },
  contentContainer: {
    flex: 1,
  },
  text: {
    color: 'white',
    alignSelf: 'center',
    margin: 15,
  },
  button: {
    alignSelf: 'center',
    paddingLeft: 15,
    paddingRight: 15,
    marginBottom: 15,
  },
});

class FuelRefillScreen extends React.Component {
  static navigationOptions = {
    title: 'Drivstoff',
    headerRight: <SettingsButton />,
  }

  constructor(props) {
    super(props);
    this.props.getItems();
  }

  render() {

    let modal = (<View style={styles.modal} />);

    if (this.props.user) {
      const user = JSON.parse(this.props.user);
      if (!user.FuelNotification) {
        modal = (
          <View style={styles.modal}>
            <FuelSetNotificationScreen />
          </View>
        );
      }
    }

    return (
      <View style={styles.container}>
        {modal}
        <View style={styles.contentContainer}>
          <FuelRefillForm />
          <Content contentContainerStyle={styles.content}>
            <FuelRefillList />
          </Content>
        </View>
      </View>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  getItems: () => {
    dispatch(getFuelRefills());
  },
});

const mapStateToProps = (state) => {
  return {
    user: state.auth.user,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(FuelRefillScreen);
