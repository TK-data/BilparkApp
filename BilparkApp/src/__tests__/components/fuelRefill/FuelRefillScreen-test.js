import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import FuelRefillScreen from '../../../components/fuelRefill/FuelRefillScreen';

const initialState = {
  auth: {
    user: JSON.stringify({ FuelNotification: true }),
  },
};


const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing FuelRefillScreen', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <FuelRefillScreen />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });
});
