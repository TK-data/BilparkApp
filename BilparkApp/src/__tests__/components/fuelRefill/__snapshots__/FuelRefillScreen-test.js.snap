// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FuelRefillScreen Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FuelRefillScreen
    getItems={[Function]}
    user="{\\"FuelNotification\\":true}"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "alignContent": "center",
            }
          }
        />,
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Connect(FuelRefillForm) />
          <Styled(Content)
            contentContainerStyle={
              Object {
                "backgroundColor": "#002776",
                "flex": 2,
                "justifyContent": "flex-start",
              }
            }
            virtual={undefined}
          >
            <Connect(FuelRefillList) />
          </Styled(Content)>
        </View>,
      ],
      "style": Object {
        "backgroundColor": "#002776",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "alignContent": "center",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Connect(FuelRefillForm) />,
            <Styled(Content)
              contentContainerStyle={
                Object {
                  "backgroundColor": "#002776",
                  "flex": 2,
                  "justifyContent": "flex-start",
                }
              }
              virtual={undefined}
            >
              <Connect(FuelRefillList) />
            </Styled(Content)>,
          ],
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(FuelRefillList) />,
              "contentContainerStyle": Object {
                "backgroundColor": "#002776",
                "flex": 2,
                "justifyContent": "flex-start",
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "alignContent": "center",
              }
            }
          />,
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Connect(FuelRefillForm) />
            <Styled(Content)
              contentContainerStyle={
                Object {
                  "backgroundColor": "#002776",
                  "flex": 2,
                  "justifyContent": "flex-start",
                }
              }
              virtual={undefined}
            >
              <Connect(FuelRefillList) />
            </Styled(Content)>
          </View>,
        ],
        "style": Object {
          "backgroundColor": "#002776",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "alignContent": "center",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Connect(FuelRefillForm) />,
              <Styled(Content)
                contentContainerStyle={
                  Object {
                    "backgroundColor": "#002776",
                    "flex": 2,
                    "justifyContent": "flex-start",
                  }
                }
                virtual={undefined}
              >
                <Connect(FuelRefillList) />
              </Styled(Content)>,
            ],
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Connect(FuelRefillList) />,
                "contentContainerStyle": Object {
                  "backgroundColor": "#002776",
                  "flex": 2,
                  "justifyContent": "flex-start",
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
